### PR TITLE
deploys android and androidTV together

### PR DESCRIPTION
**Pull request by redreceipt**
Screenshot 2022-11-23 at 11 58 01 PM
Screenshot 2022-11-23 at 11 58 05 PM

### CODE REVIEW

**Comment by redreceipt** [comment]
still need to test this locally

**Comment by vinnyjth** [comment]
This is awesome 🙌 

Will this result in a TV app being installable for all Android apps by default?

**Comment by redreceipt** [comment]
that's the goal

**Comment by vinnyjth** [comment]
> that's the goal

Gotcha. We might want to reconsider that (we should be able to avoid shipping a TV app). 

The majority of churches won't have the correct channels setup in order to provide a TV experience, and we don't prohibit shipping updates to those churches until we set them up with TV channels.

**Comment by redreceipt** [comment]
yea I should be able to set an on/off switch

**Comment by redreceipt** [comment]
ok this is ready to test locally but due to the high stakes situation of Android version codes, I wanna make sure I get the algorithm right before trying 😂 @vinnyjth @calebpanza double check my code and algorithm here. And if you have a better suggestion, I'm open to it. The version code is hours since Unix epoch with a 1/2 in front depending on mobile/tv. And then all of that added to an optional offset. So for Newspring it would be:

```
  2022000000 (offset)
+    1000000 (1 for mobile)
+     463200 (hours since unix epoch)
------------
  2023463200
```

TV would be `2024463200`

I also made the iOS build number the hours since the epoch to match. But it's almost irrelevant.

**Comment by calebpanza** [comment]
>   2022000000 (offset)
+    1000000 (1 for mobile)
+     463200 (hours since unix epoch)
------------
  2023463200

So when the year turns, we'll update this to 2023?

**Comment by redreceipt** [comment]
> First off, this looks great! I feel like this is making the most out of an android situation. Lol.
> 
> I know there's only so much that we can do, but is there any cap to the build number that we should be aware of? The offsets are huge which is great, but are we at all concerned that we could possibly hit a "cap" that is hidden in the Android docs somewhere?

`210000000` is the cap 😑

**Comment by redreceipt** [comment]
Not too concerned about hitting it because, if my math is right, the hours won't dip into the 7th digit for another 50 years

**Comment by redreceipt** [comment]
The draw back is we have a one hour release timetable. Even moving to minutes put us too close to the cap.

File: templates/tv/ios/Podfile.lock [hunk 1]
```diff
@@ -457,4 +457,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: b8198b053d33b10a2bf51f86e92e550936c6a83a
 
-COCOAPODS: 1.11.3
```
[the remark this answers is not in the document]
Comment (redreceipt) [reply templates/tv/ios/Podfile.lock:460]: that's the version of cocoa pods listed the `Gemfile` not sure why it was wrong to begin with.

**Comment by calebpanza** [comment]
> The draw back is we have a one hour release timetable. Even moving to minutes put us too close to the cap.

I don't _love_ that, but as long as it's well documented and known that it's a limitation, then that's fine. It's takes a good 30 minutes to push a release anyway.

Maybe this is something that @nlewis84 could add as a consideration into the Deployment Dashboard?

**Comment by vinnyjth** [comment]
Ah! so the goal here is that churches that have both an Android and TV app would run a different script to deploy both apps?  Churches that lack TV would continue to run the same script?

**Comment by redreceipt** [comment]
Exactly

**Comment by vinnyjth** [comment]
This looks great to me - 

My expectation is that when deploying to TV and Mobile, we would end up with two bundles in Google Play Console for the same bundle ID. One would be valid for a certain set of devices (mobile) and one would be valid for a different set of devices (TV). We would promote both to production. Does that sound right? 

Want me to run a test on Bitrise before deploying this or should we yolo and go straight to `master` and test from there?

**Comment by vinnyjth** [comment]
Also happy to have you on board with the `timestamps as build numbers` bandwagon ;)

**Comment by redreceipt** [comment]
likely blocked until we can get this patch into tv somehow

**Comment by redreceipt** [comment]
@dsilva609 may have a `build.gradle` patch to unblock this?

**Comment by dsilva609** [comment]
> @dsilva609 may have a `build.gradle` patch to unblock this?

Opened https://github.com/ApollosProject/apollos-apps/pull/2764 to fix the build

**Comment by jordanwade** [comment]
@redreceipt do you need any help getting this one ready to review again and merged?

**Comment by redreceipt** [comment]
I was sort of waiting on icons to be right to merge this but I suppose they are unrelated and can be fixed separately. I'll open this one up this afternoon and let you know when it's ready to review. sorry everything is held up. the path for all this feels very linear.

**Comment by redreceipt** [comment]
it worked!

File: templates/mobile/android/app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -6,6 +6,10 @@
     <uses-permission android:name="android.permission.CAMERA" />
     <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION"/>
     <uses-permission android:name="android.permission.FOREGROUND_SERVICE" />
+    <supports-screens android:smallScreens="true"
```
Comment (vinnyjth) [inline templates/mobile/android/app/src/main/AndroidManifest.xml:9]: These might not be the right values :) 

Here's what Crossroads discovered: 

https://github.com/Differential/crossroads-anywhere/pull/6808

https://github.com/Differential/crossroads-tv/pull/44

**Review by vinnyjth — CHANGES_REQUESTED** [review]

File: templates/mobile/android/app/src/main/AndroidManifest.xml (hunk 2 above)
Comment (redreceipt) [reply templates/mobile/android/app/src/main/AndroidManifest.xml]: good find. the tv size thing always seemed a little thin to me. fixed.

Comment (calebpanza) [reply templates/mobile/android/app/src/main/AndroidManifest.xml]: @redreceipt idk if I'm missing it or not, but it looks like the TV Manifest files were updated but not the Mobile ones per the [PR reference to the Crossroads Mobile App](https://github.com/Differential/crossroads-anywhere/pull/6808/files)

Comment (redreceipt) [reply templates/mobile/android/app/src/main/AndroidManifest.xml]: I'm not sure what's going on with the crossroads repo but the mobile _does_ require a touchscreen, no?

Comment (redreceipt) [reply templates/mobile/android/app/src/main/AndroidManifest.xml]: it's TV where, touchscreen should be false. that's straight from the docs https://developer.android.com/training/tv/start/start#no-touchscreen

Comment (calebpanza) [reply templates/mobile/android/app/src/main/AndroidManifest.xml]: Yeah I'm not too sure about the touch screen bit. I had the same thought you mentioned. Maybe their app use to be allowed on the TV store?

As for the device sizes, I don't see any specified in our application? I'm looking at `templates/mobile/android/app/src/main/AndroidManifest.xml` both on `master` and this branch and I don't see any specification

Comment (redreceipt) [reply templates/mobile/android/app/src/main/AndroidManifest.xml]: I'm assuming specifying sizes aren't necessary with using the touchscreen and TV booleans. It's not mentioned in the developer docs and crossroads just sets all sizes to true which seems like an unnecessary smell.

**Review by calebpanza — APPROVED** [review]

**Review by vinnyjth — APPROVED** [review]

**Pull request merged**